src/renderer/lib/hooks/useDolphin.ts

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable react-hooks/rules-of-hooks */
```

Comment:
Why are these eslint rules disabled?

Reply:
that's a leftover from when I was trying to write the hook in a different way, I'll remove it.